### PR TITLE
[AzCopyV10][Feature] Add mirror mode to sync

### PR DESCRIPTION
**Pull request by mohsha-msft**

### CODE REVIEW

File: cmd/sync.go [hunk 1]
```diff
@@ -64,6 +64,9 @@ type rawSyncCmdArgs struct {
 	// otherwise the user is prompted to make a decision
 	deleteDestination string
 
+	// this flag is to disable comparator and overwrite files at destination irrespective
+	mirrorMode bool
```
Comment (mohsha-msft) [inline cmd/sync.go:68]: I'll happily change this name to something else.

**Review by nakulkar-msft — APPROVED** [review]

**Review by adreed-msft — APPROVED** [review]

File: cmd/sync.go (hunk 1 above)
Comment (nakulkar-msft) [reply cmd/sync.go:68]: Mirror mode - as far as this change goes- does copy alone, and you'll have to additionally use delete-destination with --mirrorMode to achieve what we think is intuitional.

**Comment by Kapanther** [comment]
I guess the difference between `mirror` and `overwrite` is context between a "`sync`" and a "`copy`"

Does `overwrite=true` skip files with the same Date modified Stamps? I'm guessing not... But not sure.

Where as mirror-mode in a `sync` context still performs the date modified check, but will not overwrite files that have the same file stamp. Just files with older or newer stamps.

**Comment by phrak** [comment]
As a long-time robocopy user, "Mirror" mode should be consistent with the robocopy behaviour.
Mirror = Copy + Delete from destination + Overwrite if different.

AZCopy still has a major functionality gap, not having an "Overwrite if different" capability. 
Hopefully the new "Mirror" mode will be able to handle this scenario.

**Pull request merged**